### DISCUSSION TITLE
Rebuild Jellyfin server in rust

### DISCUSSION THREAD

**Discussion by Uziniii** [jellyfin/jellyfin#10923] [Polls]
I find Jellyfin very slow and i think is cause of the server

**Comment by nielsvanvelzen** [comment]
Are you planning to rewrite the entire server or is this just another "C# bad, rust good" circlejerk post?

**Reply by Uziniii** [reply]
While there are no current plans to rewrite the entire server, I'm not planning to start it right away; maybe I'll consider doing it much later. I don't believe C# is bad, but Rust is known for its strengths in multithreading, which could be beneficial for tasks like indexing all movies and TV series

Or it might be interesting to know why Jellyfin seems slow.

**Reply by sjorge** [reply]
I don't think it makes much sense to rewrite Jellyfin in a different language, finding and fixing issues in the current code base is going to give better and more tangible results.

This sort of comes to mind: https://xkcd.com/927/

There is also already a [project](https://github.com/Dusk-Labs/dim) implementing a media server in Rust, which is not very fast either from my brief time poking it.

**Reply by Uziniii** [reply]
Maybe the problem here come from the frontend
Each time we scroll, images are just refreshing

**Comment by crobibero** [comment]
No.

**Comment by joshuaboniface** [comment]
To mirror the rest of the responses, no, this is not going to happen.

This is something that gets "suggested"/brought up every 6 months or so. The simple fact is this: starting a project as in-depth as Jellyfin from scratch (in a new language or not) is a LOT of work. You'd need to put in an immense amount of work to even get to a basic, core featureset. You need to make many dozens of decisions picking answers that will both appeal to developers (language wars, yay) and users. Numerous people have tried, and not one of them has gotten off the ground.

So, this is not on anyone's radar. The reason Jellyfin took off, or even exists to begin with, is the fact that that we were able to start with a (mostly) functional codebase from Emby that had enough core features that people could use it day-to-day, and were able to build on it from there. Was it in the "best" language, with the best most readable codebase, with every feature? No. But the bones were there, in a popular *enough* language to get volunteer contributors to contribute, and it went from there.

We're never going to discourage anyone from doing this; if you want to, go for it. But ensure you understand exactly what you're getting into. It's a fair bit beyond a one-person hobby project.
